### PR TITLE
chore: bump python to 3.9

### PR DESCRIPTION
**Pull request by Anas12091101**
### What are the relevant tickets?
https://github.com/mitodl/hq/issues/4303

### Description (What does it do?)

This PR updates the python version of edx-api-client to 3.9.x

### How can this be tested?
- Make sure your edx-platform and xpro both are running.
- Configure your xpro with openedx following the steps mentioned in [configure_open_edx.md](https://github.com/mitodl/mitxpro/blob/master/docs/configure_open_edx.md)
- Run all the management commands using the edx-api-client and validate that everything's working fine

### CODE REVIEW

**Comment by Anas12091101** [comment]
I setup tutor and configured edx-platform with xpro. I was getting error: `Session value state missing.`. When I set `SOCIAL_AUTH_REDIRECT_IS_HTTPS = True`, my redirect uris also become `https` endpoints due to which I couldn't test 
 it locally. I have tested `get_edx_api_service_client` which is working. But `get_valid_edx_api_auth` in `get_edx_api_client()` method is throwing this error.

**Comment by arslanashraf7** [comment]
@mudassir-hafeez Could you review this PR?

File: edx_api/grades/__init__.py [hunk 1]
```diff
@@ -103,7 +100,7 @@ def get_course_current_grades(self, course_id):
                 resp = self.requester.get(resp_json['next'])
                 resp.raise_for_status()
                 resp_json = resp.json()
-                grade_entries.extend((CurrentGrade(entry) for entry in resp_json["results"]))
+                grade_entries.extend(CurrentGrade(entry) for entry in resp_json["results"])
         else:
```
Comment (mudassir-hafeez) [inline edx_api/grades/__init__.py:104]: Is there any specific reason for making this change?

File: edx_api/enrollments/init_test.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 from unittest import TestCase
 
 try:
-    from mock import patch
+    from unittest.mock import patch
 except ImportError:
     from unittest.mock import patch
```
Comment (mudassir-hafeez) [inline edx_api/enrollments/init_test.py:11]: I can see two imports being used.

**Review by mudassir-hafeez — CHANGES_REQUESTED** [review]
Overall, this looks good 👍. I just have a few minor requests.
I'm currently testing this PR.

File: edx_api/grades/__init__.py (hunk 1 above)
Comment (Anas12091101) [reply edx_api/grades/__init__.py:104]: The python codemod [pyupgrade](https://github.com/asottile/pyupgrade) changed this itself. I think, in previous python versions, a bracket would be required to write such expressions which is not necessary now. Is it breaking anything?

**Review by mudassir-hafeez — APPROVED** [review]
I've tested with Xpro. LGTM!

**Pull request merged**